consumer/transaction.go

```diff
@@ -270,12 +302,26 @@ func txnConsume(s *shard, txn *transaction, env message.Envelope) error {
 		s.clock.Update(txn.beganAt.Add(time.Duration(delta)))
 	}
 	txn.consumedCount++
-	txn.consumedBytes += (env.End - env.Begin)
+	txn.consumedBytes += (s.sequencer.Dequeued.End - s.sequencer.Dequeued.Begin)
```

Comment:
Seems like maybe we should wait to update these stats until after we know the message isn't getting deferred until the next transaction.

Reply:
Done.